### ISSUE TITLE
Align with official figma types?

### ISSUE THREAD

**Issue by gossi** [didoo/figma-api#34]
I've been developing a figma plugin at first and later on a tool to connect to figma, which led me here. For figma plugin developers, there are official types as [`@figma/plugin-typings`](https://github.com/figma/plugin-typings). Using them felt intuitive while following their docs.

So, when using the types from this repo it felt a bit _awkward_ as this is `Node<'TEXT'>` (which I am still using?!?) vs `TEXT` vs `TextNode` (which it is for figma plugins).

I was wondering, if they somehow can be aligned, as the properties from figma plugins are pretty much equal to what is available through the REST-API (or a subset). So, at least the _names_ to make them sound equal?

As for the subset, if possible, they may even be picked? such as: `type VectorNode = Pick<FigmaVectorNode, '...'>`.

PS. That may even be a complete non-sense, time-wasting thing, ready to discard (and I'm totally fine with that). But it is in my head for a little while and I wanted to share it.

**Comment by gossi** [comment]
Yes, sure.

Given these things:

- I'm an author of a figma plugin, that uses the official types from [`@figma/plugin-typings`](https://github.com/figma/plugin-typings). 
- I'm also a direct user of [AST Types](https://github.com/didoo/figma-api/blob/main/src/ast-types.ts) (instead of the `Figma.Api`). That is I'm fetching the figma file myself and then use the types from this project for my code.

Now, I'm using _two_ different (type) libraries to express the _same_ code - as the figma plugin API and the response from the figma rest API are pretty much equal (the same for my use-case). BUT the types are different. Let's make an example for a text node:

- Official types: [`TextNode`](https://github.com/figma/plugin-typings/blob/602b3a5845a67b8e75bf080c9695c43a3038735b/plugin-api.d.ts#L1110)
- This library: [`TEXT`](https://github.com/didoo/figma-api/blob/81a44f2d42b3b56ca0918c1e69d30a4bce7ee92d/src/ast-types.ts#L839) or [`Node<'TEXT'>`](https://github.com/didoo/figma-api/blob/81a44f2d42b3b56ca0918c1e69d30a4bce7ee92d/src/ast-types.ts#L916)

This confuses me 😵 - so I was wondering, if the types from this library can/should be renamed to match the name from the official plugin api, so from `TEXT` -> `TextNode`.

Schematic code:

```ts
// figma-api/types.ts

export type TextNode = VectorNode & {
  // ...
}
```

been used as:

```ts
import { TextNode } from 'figma-api';

interface MyOwnLibrary {
  /** a text field */
  textField: TextNode;
}
```

My second idea (to use the official types under the hood) is kinda not possible as these types are not importable.

The idea is a combination of an aesthetic request (in my eyes 🙈) and DX improvements for folks working in a figma plugin and with the rest api. A possible migration is to allow both APIs, mark the existing one deprecated and remove it with the next major release. The effort of course is heavily questionnable.

I hope it is clearer?

**Comment by didoo** [comment]
hey @gossi I have just completely refactored the library (see #73) to align as much as possible with the official Figma REST API endpoints (https://www.figma.com/developers/api#intro) and to use https://github.com/figma/rest-api-spec under the hood (so the types signatures should be the official ones).

I just released the `2.0.0-beta` version of the library, in case you want to test it.
